src/main/java/org/sonarlint/intellij/core/AnalysisRequirementNotifications.java

```diff
@@ -24,25 +24,96 @@
 import com.intellij.notification.NotificationGroup;
 import com.intellij.notification.NotificationListener;
 import com.intellij.notification.NotificationType;
+import com.intellij.openapi.options.ShowSettingsUtil;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.wm.ToolWindow;
 import com.intellij.openapi.wm.ToolWindowManager;
 import com.intellij.ui.content.Content;
 import com.intellij.ui.content.ContentManager;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import javax.annotation.Nullable;
 import javax.swing.event.HyperlinkEvent;
 import org.jetbrains.annotations.NotNull;
+import org.sonarlint.intellij.config.global.SonarLintGlobalConfigurable;
 import org.sonarlint.intellij.ui.SonarLintToolWindowFactory;
+import org.sonarsource.sonarlint.core.client.api.common.Language;
+import org.sonarsource.sonarlint.core.client.api.common.PluginDetails;
+import org.sonarsource.sonarlint.core.client.api.common.SkipReason;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.AnalysisResults;
 
 public class AnalysisRequirementNotifications {
 
-  private static final NotificationGroup ANALYSIS_PROBLEM_GROUP = NotificationGroup.balloonGroup("SonarLint: Analysis Requirement");
+  private static final NotificationGroup ANALYZER_REQUIREMENT_GROUP = NotificationGroup.balloonGroup("SonarLint: Analyzer Requirement");
+
+  private static final Set<String> alreadyNotified = new HashSet<>();
```

Comment:
This means that users will have to reload SonarLint / restart the IDE to have the checks performed again, right?

Let's imagine someone has the warning about Node, installs it while the IDE is running (e.g with a package manager that puts it on the `$PATH`); then technically the requirement is met, but SonarLint will only notice it on restart.

I believe we have to either:
- mention that the IDE must be restarted in the message about Node
- make sure to clean this cache when the path to Node is updated in the settings

Reply:
Cache cleaned in the next commit, when the setting is updated